### PR TITLE
fix(#6604): Deno.setRaw shouldn't panic on ENOTTY

### LINKED ISSUE #6604: Issue with Deno.setRaw and subprocesses
It seems like after using Deno.setRaw, even after setting it to false, and then doing the same thing in a subprocess, deno panics. I created an example script here: https://gist.github.com/Yamboy1/c9cb9e215d3a3433193044d96a62f62a.

```
$ deno run --unstable -A https://gist.githubusercontent.com/Yamboy1/c9cb9e215d3a3433193044d96a62f62a/raw/46680c56316343ee1206fcee7168242d23eb346b/test.ts
```

### PR DESCRIPTION
**Pull request by uki00a**
Closes #6604

- `Deno.setRaw` now doesn't panic on `ENOTTY`.
- `setRawShouldNotPanicOnNoTTYContext` case is currently ignored on Windows.

```shell
$ cargo run -- run --unstable -A https://gist.githubusercontent.com/uki00a/f6b4a1511d86950b3469aace1cd03568/raw/8048f478dc8a7a980fe8b001858a32703d2749e1/test.ts
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
     Running `target/debug/deno run --unstable -A 'https://gist.githubusercontent.com/uki00a/f6b4a1511d86950b3469aace1cd03568/raw/8048f478dc8a7a980fe8b001858a32703d2749e1/test.ts'`
this should print twice, and the panic should be after the second print
this should print twice, and the panic should be after the second print
error: Uncaught BadResource: ENOTTY: Not a typewriter
    at unwrapResponse ($deno$/ops/dispatch_json.ts:42:11)
    at Object.sendSync ($deno$/ops/dispatch_json.ts:69:10)
    at Object.setRaw ($deno$/ops/tty.ts:8:3)
    at https://gist.githubusercontent.com/uki00a/f6b4a1511d86950b3469aace1cd03568/raw/8048f478dc8a7a980fe8b001858a32703d2749e1/test.ts:3:6
```
https://gist.github.com/uki00a/f6b4a1511d86950b3469aace1cd03568

### CODE REVIEW

File: cli/js/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ declare namespace Deno {
     BadResource: ErrorConstructor;
     Http: ErrorConstructor;
     Busy: ErrorConstructor;
+    NoTTY: ErrorConstructor;
```
Comment (ry) [inline cli/js/lib.deno.ns.d.ts:43]: Why not use BadResource instead?

**Comment by ry** [comment]
@uki00a I've done a few adjustments to your patch. Does it look ok?

**Comment by uki00a** [comment]
@ry Yes. It looks good to me. Thanks!

**Comment by uki00a** [comment]
@ry Thanks for your review and help!

**Pull request merged**